src/orchestrator.ts

```diff
@@ -53,48 +65,204 @@ export class Orchestrator {
                     actions.push(partialResult.actions);
                 }
 
-                if (!partialResult.isCompleted) {
+                if (this.shouldFinish(partialResult)) {
                     this.finish(context, state, actions);
                     return;
+                } else if (partialResult instanceof Task && partialResult.isFaulted) {
+                    gen.throw(partialResult.exception);
                 }
 
                 decisionStartedEvent = state.find((e) =>
                     e.EventType === HistoryEventType.OrchestratorStarted &&
                     e.Timestamp > decisionStartedEvent.Timestamp);
                 context.df.currentUtcDateTime = decisionStartedEvent.Timestamp;
             } catch (error) {
-                this.error(context, error);
+                this.error(context, actions, error);
                 return;
             }
         }
     }
 
-    private callActivityAsync(state: HistoryEvent[], name: string, input: any = "__activity__default") {
+    private callActivity(state: HistoryEvent[], name: string, input?: any) {
```

Comment:
I think this is a case for any. At this point the input hasn't been serialized yet, so it could legally be a(n):

- boolean
- string
- number
- object
- array
- tuple
- enum
- undefined (I don't know why you would, but you could)
- null (we allow passing null to CallActivityAsync in the C# API)

Reply:
If you're targeting 3.0 TypeScript, you can use `Unknown` for these types, which forces type sniffing down stream.